File: Harbour Space/D_Backspace.cpp
#include<bits/stdc++.h>

using namespace std;

//typedef long long int;
#define inf 1e18
#define mod 1000000007
#define pb push_back



int main(int argc, char const *argv[])
{
    int T;
    cin>>T;
    
    for(int l=0; l<T; l++){
        string s;
        cin>>s;
        
        string t;
        cin>>t;
        
        int n = s.size();
        int m = t.size();

        if(m>n){
            cout<<"NO"<<endl;
            continue;
        }

        if(t=="" or s==t){
            cout<<"YES"<<endl;
            continue;
        }

        bool prev[m+1] = {0};
        bool curr[m+1] = {0};
        curr[0]=true;
        prev[0]=true;

        for (int i=1;i<=n; i++){
            for (int j=1;j<=m; j++){
                int q = curr[j];
                if(s[i-1]==t[j-1])
                    curr[j] = prev[j-1];
                else{
                    if(i>=2)
                        curr[j] = prev[j];
                    else
                        curr[j] = false;
                }
                prev[j]=q;
            }
        }
        

        if(curr[m])
             cout<<"YES"<<endl;
        else
             cout<<"NO"<<endl;
    }
    return 0;
}

File: Harbour Space/D_Backspace.py
for _ in range(int(input())):
    s = input()
    t = input()

    n = len(s)
    m = len(t)

    if(m>n):
        print("NO")
        continue

    if(t=="" or s==t):
        print("YES")
        continue

    i=n-1
    j=m-1
    while(i>=0 and j>=0):
        if(s[i]==t[j]):
            i-=1
            j-=1
        else:
            i-=2
    if(j<0):
        print("YES")
    else:
        print("NO")
